Documentation/EtcdLease.md

```diff
@@ -0,0 +1,38 @@
+# Overview
+The EtcdLease implement interface `grant` and `revoke` to new a lease and revoke a lease to etcd.
+It can manage the leases to send keep alive request for leases.
+
+
+## `startKeepAliveService` function
+..* The function will start a backgroud scheduler `keepAliveSchedule` to send keep alive request for lease registered to EtcdLease Client.
+..* It will open a StreamObserver to etcd, send request to this StreamObserver can keep the lease alive
+..* It will new a ScheduledExecutorService to run `keepAliveExecutor` and `deadLineExecutor` periodically
+
+## `closeKeepAliveService` function
+..* This function will end the backgroud scheduler.
+..* It will close the StreamObserver to etcd
+
+## `keepAlive` function
+..* It will new a Lease object to keep the lease id and nextKeepAliveTime and DeadLine
+..* It will add the lease object to keepAlives map
+..* the backgroud scheduler will send keep alive request for the registered leases. 
+
+## `keepAliveExecutor` function
+..* This function is called periodically by `keepAliveSchedule`
+..* It will scan the keepAlives map and get the leases that the nextKeepAliveTime before now
```

Comment:
It will scan the keepAlives map and find the leases that requires keepAlive requests to send based on its nextKeepAliveTime